sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1045,7 +1044,18 @@ static void _handle_nbr_sol(gnrc_netif_t *netif, const ipv6_hdr_t *ipv6,
                     break;
             }
         }
-        reply_aro = _copy_and_handle_aro(netif, ipv6, nbr_sol, aro, sl2ao);
+        gnrc_pktsnip_t *reply_aro = NULL;
+        if (aro && sl2ao) {
+        /* If no SLLAO is included, then any included ARO is ignored. */
+            if (!(reply_aro = _copy_and_handle_aro(netif, ipv6, nbr_sol, aro, sl2ao))) {
+            /* If the Length field is not two, or if the Status field is not zero,
+               then the NS is silently ignored.*/
+                return;
+            }
+        }
+        else if (sl2ao) {
+            _handle_sl2ao(netif, ipv6, (const icmpv6_hdr_t *)nbr_sol, sl2ao);
+        }
         /* check if target address is anycast */
```

Comment:
```suggestion
        if (reply_aro == NULL) {
            return;
        }
        /* check if target address is anycast */
```

Reply:
```C
        gnrc_pktsnip_t *reply_aro = NULL;
        if (aro && sl2ao) {
        /* If no SLLAO is included, then any included ARO is ignored. */
            reply_aro = _copy_and_handle_aro(netif, ipv6, nbr_sol, aro, sl2ao);
            /* If the Length field is not two, or if the Status field is not zero,
               then the NS is silently ignored.*/
        }
        else if (sl2ao) {
            _handle_sl2ao(netif, ipv6, (const icmpv6_hdr_t *)nbr_sol, sl2ao);
        }
        if (reply_aro == NULL) {
            return;
        }
        /* check if target address is anycast */
        if (netif->ipv6.addrs_flags[tgt_idx] & GNRC_NETIF_IPV6_ADDRS_FLAGS_ANYCAST) {
            _send_delayed_nbr_adv(netif, &nbr_sol->tgt, ipv6, reply_aro);
        }
        else {
            gnrc_ndp_nbr_adv_send(&nbr_sol->tgt, netif, &ipv6->src,
                                  ipv6_addr_is_multicast(&ipv6->dst),
                                  reply_aro);
        }
```
Did I construct this right?
Because this would mean that we would not answer an NS if the ARO has errors, which is correct, but also if there is no `ARO` at all, which is not correct.